internal/worker/secretspruner/worker.go

```diff
@@ -4,18 +4,23 @@
 package secretspruner
 
 import (
+	"strconv"
+	"strings"
+
+	"github.com/juju/collections/set"
 	"github.com/juju/errors"
 	"github.com/juju/worker/v4"
 	"github.com/juju/worker/v4/catacomb"
 
 	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/secrets"
 	"github.com/juju/juju/core/watcher"
 )
 
 // SecretsFacade instances provide a set of API for the worker to deal with secret prune.
 type SecretsFacade interface {
-	WatchRevisionsToPrune() (watcher.NotifyWatcher, error)
-	DeleteObsoleteUserSecrets() error
+	WatchRevisionsToPrune() (watcher.StringsWatcher, error)
+	DeleteObsoleteUserSecrets(uri *secrets.URI, revisions ...int) error
```

Comment:
I think the method name should be `DeleteObsoleteUserSecretRevisions` as we are just deleting selected revisions, not the entire secret